### ISSUE TITLE
Windows Installer: Remove WSL

### ISSUE THREAD

**Issue by mook-as** [rancher-sandbox/rancher-desktop#263]
We should augment the Windows installer to remove WSL when we uninstall the app, provided that
- We installed WSL in the first place
- There are no other distributions registered

We should also unregister the k3s distribution, regardless of whether we installed WSL.

See #260 for related work (installing WSL).

**Comment by jandubois** [comment]
I'm not convinced that this is really necessary, and even if all your prerequisite conditions are met, I think this would still require an explicit confirmation from the user that they want to remove WSL as well.

The same way we need to query if the user wants to keep the cache of downloaded k3s versions around. Maybe the uninstall is just temporary, to resolve some issue, and the user wants to reinstall again in the near future.

**Comment by overshareware** [comment]
> I'm not convinced that this is really necessary, and even if all your prerequisite conditions are met, I think this would still require an explicit confirmation from the user that they want to remove WSL as well.
> 
> The same way we need to query if the user wants to keep the cache of downloaded k3s versions around. Maybe the uninstall is just temporary, to resolve some issue, and the user wants to reinstall again in the near future.

Agreed, removing WSL isn't necessary, but it would be nice if the rancher-desktop distros weren't left hanging around after uninstall..

**Comment by jandubois** [comment]
As of v1.17.1, the installer no longer installs WSL2 at all, which removes the original premise (uninstalling WSL that RD installed). The narrower concern — cleaning up RD-specific WSL distros on uninstall — is tracked in #5380. Removing WSL itself on uninstall would require careful user confirmation and could be harmful for users doing a reinstall.
